scripts/js/lib/links/InternalLink.ts

```diff
@@ -114,14 +114,14 @@ export class InternalLink {
     let suggestionPath: String = "";
     let suggestionPathAnchors: string[] = [];
 
-    const possiblePaths = this.possibleFilePaths(originFile);
-    const pathNoExtension = possiblePaths[0].replace(/\.[^\/.]+$/, "");
-
     existingFiles.forEach((file) => {
-      let score = levenshtein.get(pathNoExtension, file.path);
+      const candidatePath = file.path.startsWith("public/")
+        ? file.path.replace(/^public/, "")
+        : file.path.replace(/\.[^\/.]+$/, "").replace(/^docs/, "");
+      let score = levenshtein.get(this.value, candidatePath);
```

Comment:
`this.value` is the problematic link. The anchor is stripped and stored separately in `this.anchor`.